src/script/components/modal.ts

```diff
@@ -31,42 +33,79 @@ interface ModalParams {
 
 ko.components.register('modal', {
   template: `
-    <div class="modal" data-bind="style: {display: displayNone() ? 'none': 'flex', zIndex: 10000001}">
+    <div class="modal" data-bind="style: {display: displayNone() ? 'none': 'flex', zIndex: 10000001}, attr: id ? {id: id, 'aria-labelledby': ariaLabelby} : {}" role="dialog" aria-modal="true" tabindex="-1">
       <!-- ko if: showLoading() -->
         <loading-icon class="modal__loading"></loading-icon>
       <!-- /ko -->
       <!-- ko ifnot: showLoading() -->
-        <div class="modal__content" data-bind="css: {'modal__content--large': large, 'modal__content--visible':  hasVisibleClass() && !showLoading()}, fadingscrollbar" >
+        <div class="modal__content" data-bind="css: {'modal__content--large': large, 'modal__content--visible':  hasVisibleClass() && !showLoading()}, fadingscrollbar">
           <!-- ko template: { nodes: $componentTemplateNodes, data: $parent } --><!-- /ko -->
         </div>
       <!-- /ko -->
     </div>
-    <div class="modal__overlay" data-bind="click: () => onBgClick(), css: {'modal__overlay--visible': hasVisibleClass()}, style: {display: displayNone() ? 'none': 'flex'}" ></div>
+    <div class="modal__overlay" data-bind="click: () => onBgClick(), css: {'modal__overlay--visible': hasVisibleClass()}, style: {display: displayNone() ? 'none': 'flex'}"></div>
     `,
   viewModel: function ({
     isShown,
     large,
+    ariaLabelby,
     onBgClick = noop,
     onClosed = noop,
     showLoading = ko.observable(false),
   }: ModalParams): void {
     this.large = large;
+    this.id = createRandomUuid();
+    this.ariaLabelby = ariaLabelby;
     this.onBgClick = () => ko.unwrap(onBgClick)();
     this.displayNone = ko.observable(!ko.unwrap(isShown));
     this.hasVisibleClass = ko.computed(() => isShown() && !this.displayNone()).extend({rateLimit: 20});
     this.showLoading = showLoading;
     let timeoutId = 0;
+
+    const maintaineFocus = (): void => {
+      if (!this.displayNone()) {
+        document.addEventListener('keydown', this.onKeyDown);
+      }
+    };
+
+    this.onKeyDown = (event: KeyboardEvent): void => {
```

Comment:
Not need to add the `onKeyDown` function to `this` since it's not accessed anywhere outside of this scope .

You could just do `const onKeyDown = ...` and remove the `this` on `(add|remove)EventListener`